Userland/Libraries/LibWeb/HTML/Window.cpp

```diff
@@ -496,11 +496,11 @@ void Window::fire_a_page_transition_event(FlyString const& event_name, bool pers
 void Window::queue_microtask_impl(WebIDL::CallbackType& callback)
 {
     // The queueMicrotask(callback) method must queue a microtask to invoke callback,
-    HTML::queue_a_microtask(&associated_document(), [&callback]() mutable {
+    HTML::queue_a_microtask(&associated_document(), [this, &callback]() mutable {
         auto result = WebIDL::invoke_callback(callback, {});
         // and if callback throws an exception, report the exception.
         if (result.is_error())
-            HTML::report_exception(result);
+            HTML::report_exception(result, realm());
```

Comment:
Same concern here, should we capture a weak pointer to this using `make_weak_ptr<Window>()` and checking it for GC before calling window->realm()?

Reply:
I suppose if a window object was GC'd than invoking the callback doesn't make sense. It seems to work:

<img width="557" alt="Screen Shot 2022-10-15 at 00 19 28" src="https://user-images.githubusercontent.com/26363017/195937505-c9fba02a-bcb2-4b4f-8760-cf7f5bc33d30.png">